### PR TITLE
Add RsTreeWidget Sort Context Menu

### PR DESCRIPTION
**Pull request by PhenomRetroShare**
Now Header sort when visible or In context menu is hidden.

### CODE REVIEW

File: retroshare-gui/src/gui/common/GroupTreeWidget.cpp [hunk 1]
```diff
@@ -167,8 +160,8 @@ void GroupTreeWidget::addToolButton(QToolButton *toolButton)
 
 	/* Initialize button */
 	toolButton->setAutoRaise(true);
-	toolButton->setIconSize(ui->displayButton->iconSize());
-	toolButton->setFocusPolicy(ui->displayButton->focusPolicy());
+	toolButton->setIconSize(QSize(24,24));
```
Comment (csoler) [inline retroshare-gui/src/gui/common/GroupTreeWidget.cpp]: Bad way to do it: you need 24*QFontMetricsF(font()).height()/14.0, so it remains proportional to font size and keeps the same look on high DPI screens.

File: retroshare-gui/src/gui/common/RSTreeWidget.cpp [hunk 2]
```diff
@@ -283,13 +283,57 @@ QMenu *RSTreeWidget::createStandardContextMenu(QMenu *contextMenu)
 		actShowHeader->setCheckable(true);
 		actShowHeader->setChecked(!isHeaderHidden());
 
-		QMenu *headerMenu = contextMenu->addMenu(QIcon(),tr("Show column..."));
-
 		QTreeWidgetItem *item = headerItem();
 		int columnCount = item->columnCount();
+
```
Comment (csoler) [inline retroshare-gui/src/gui/common/RSTreeWidget.cpp:290]: Is this generic enough to work with all different instances of RSTreeWidget? It's used quite a lot.

File: retroshare-gui/src/gui/common/GroupTreeWidget.cpp (hunk 1 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/common/GroupTreeWidget.cpp]: OK,
For next, could you test with this:
The Qt::AA_EnableHighDpiScaling application attribute, introduced in Qt 5.6, enables automatic scaling based on the monitor's pixel density.
https://doc.qt.io/qt-5/highdpi.html

File: retroshare-gui/src/gui/common/RSTreeWidget.cpp (hunk 2 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/common/RSTreeWidget.cpp:290]: Yes

File: retroshare-gui/src/gui/common/GroupTreeWidget.cpp (hunk 1 above)
Comment (PhenomRetroShare) [reply retroshare-gui/src/gui/common/GroupTreeWidget.cpp]: Now, I use:
	int i = qt_defaultDpi();
	auto desktopWidget = QApplication::desktop();
	auto y = desktopWidget->logicalDpiY();

	toolButton->setIconSize(QSize(24*y/i,24*y/i));
If it's OK, we can make it static and in utils function.

**Comment by jolavillette** [comment]
thx for this PR
to fix:
- when sorting by "last post" a forum with no post should **not** appear as the most recently updated forum
- when sorting by "number of posts" or "popularity" please use numeric sort **not** alphanumeric

**Comment by PhenomRetroShare** [comment]
@jolavillette : Done...

**Pull request merged**